signing/sigstore/signer.go

```diff
@@ -115,15 +116,16 @@ func (s *Signer) Sign(ctx context.Context, p *intoto.ProvenanceStatement) (*Atte
 }
 
 // Upload uploads the signed attestation to the rekor transparency log.
-func (s *Signer) Upload(ctx context.Context, att *Attestation) error {
+func (s *Signer) Upload(ctx context.Context, att *Attestation) (*models.LogEntryAnon, error) {
```

Comment:
@lumjjb I originally thought this would become part of a generic interface that could support other signers and transparency logs besides cosign/rekor. Is there a use case you had for this?

Reply:
yea agreed - was thinking about that as well when i was making this change, but wasn't sure what the interface would look like so exposed it for now since we wanted to expose the log ID so that it can be retrieved more easily (instead of having to do a search on rekor)

Reply:
Yeah, on the other side of things I didn't have a good idea what the use cases would be so I left it off.

Where did you want to expose the ID? Maybe a more generic struct or interface with a string ID would work? (similar to the Attestation struct)